src/app/loadout-builder/generated-sets/GeneratedSet.tsx

```diff
@@ -86,9 +88,31 @@ function GeneratedSet({
     return itemModAssignments;
   }, [displayedItems, lockedMods, armorEnergyRules]);
 
-  if (set.armor.some((items) => !items.length)) {
-    errorLog('loadout optimizer', 'No valid sets!');
-    return null;
```

Comment:
This doesn't seem to do anything in practice -- if a bucket has no items, `displayedItems` will contain an `undefined` and `fitMostMods` will crash? The useMemo makes it inconvenient to check earlier.